packages/salesforcedx-vscode-core/package.json

```diff
@@ -111,6 +111,11 @@
         {
           "command": "sfdx.force.task.stop",
           "when": "view == sfdx.force.tasks.view"
+        },
+        {
+          "command": "sfdx.force.metadata.view.refresh",
+          "when": "view == metadata && viewItem == refreshable",
```

Comment:
Doesn't seem like the when clause supports OR so i had to mark the contextValue for nodes as "refreshable" that support the functionality

Reply:
Currently, the user can view the 'refresh' icon on metadata type nodes even if they haven't expanded the node to view the components yet. If you click on refresh, the refresh is executed but the component nodes still aren't displayed because the user hasn't expanded the type node. I think it can be a little unintuitive to have the ability to refresh even if the initial loading of the components hasn't occurred yet- is it possible to only display the refresh icon when the node is expanded?

Reply:
Also- it looks like the when clause doesn't support OR but it does support regex. If we would like to assign the appropriate context values for folder and type nodes (ie. 'folder' and 'type') we can set the clause to: `"when": "view == metadata && viewItem =~ /(folder|type)/"` and it should work!

Reply:
There's sort of a way to do it but it's awkward. The tree data change event has to be fired a second time which makes this stuttering effect. And even then you have to click out of the tree view and click back in to it to get the icon to actually load.